Lib/tkinter/simpledialog.py

```diff
@@ -262,7 +262,11 @@ def _setup_dialog(w):
         w.tk.call("::tk::unsupported::MacWindowStyle", "style",
                   w, "moveableModal", "")
     elif w._windowingsystem == "x11":
-        w.wm_attributes("-type", "dialog")
+        try:
+            w.wm_attributes("-type", "dialog")
+        except TclError:
```

Comment:
Rather than catching an error, would we be better checking for supported Tk version like you've done elsewhere? (`if get_tk_patchlevel() >= (8, 5, 9):`)

Reply:
There is no `get_tk_patchlevel()` in the `tkinter` module. It is only defined in tests.